### PR TITLE
poppler 0.34.0

### PR DESCRIPTION
**Pull request by bfontaine**

### CODE REVIEW

**Comment by tdsmith** [comment]
Dependents might need bumped, like in #35606.

**Comment by bfontaine** [comment]
Dependents to check/bump:

- [ ] diff-pdf
- [ ] pdf2htmlex
- [ ] pdf2svg
- [ ] pdfgrep
- [ ] pdftoipe
- [ ] xournal
- [ ] inkscape
- [ ] pdf-tools

**Comment by dunn** [comment]
Could bump poppler up to 0.32 now too.

**Comment by bfontaine** [comment]
Done.

**Comment by bfontaine** [comment]
`diff-pdf` should be migrated to homebrew/x11, @mikemcquaid should we do this now and rebase this PR or merge this one *then* migrate `diff-pdf`?

**Comment by mikemcquaid** [comment]
Looks like `pdf2htmlex` also needs a bump (and a dependency if it doesn't have one specified). Something weird going on with `mat`; looks like it needs a `python` dep either added or removed.

**Comment by bfontaine** [comment]
Bumped `pdf2htmlex` and `mat` thought I’m not done with the latter but it’s time to go to bed :sleepy:

**Comment by dunn** [comment]
Looks like the pdf2htmlex problem has been [fixed upstream](https://github.com/coolwanglu/pdf2htmlEX/issues/471); I think this is the right patch: https://github.com/coolwanglu/pdf2htmlEX/commit/d4fc82b90131470297aa67eed4aea2364933b607.diff

**Comment by bfontaine** [comment]
I added the pdf2htmlex patch to the formula. We still have to fix mat and hopefully we’ll be good :)

**Comment by luzpaz** [comment]
bump

**Comment by bfontaine** [comment]
@luzpaz “bump” comments doesn’t magically give free time to contributors to work on the issue :wink:

**Comment by dunn** [comment]
Popper 0.33.0 is out.

**Comment by bfontaine** [comment]
Thanks @dunn, I really need to get some time to work on the `mat` issue so we can merge this PR.

**Comment by DomT4** [comment]
As a heads up, the newest version of `mat` has other issues as well. I've reported it upstream but radio silence so far: #38600

**Comment by bfontaine** [comment]
`mat` is now gone (45c6a6c), hopefully we’ll now be able to merge this PR.

**Comment by mikemcquaid** [comment]
Still got a failing build: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/29092/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_pdf2htmlex/

**Comment by bfontaine** [comment]
Yes I’m aware of that but haven’t got the time to look at it (yet) :wink:

**Comment by edporras** [comment]
FYI, [poppler 0.34.0](http://poppler.freedesktop.org/poppler-0.34.0.tar.xz) is out.

**Comment by bfontaine** [comment]
Thanks @edporras, I updated the PR.

**Comment by bfontaine** [comment]
I pinged upstream re the pdf2htmlex failure.

**Comment by bfontaine** [comment]
I’m trying with a patch on `pdf2htmlex`. If the build passes I’ll submit it upstream.

**Comment by luzpaz** [comment]
@bfontaine thanks for working on this :+1:

**Comment by bfontaine** [comment]
you’re welcome @luzpaz :smile: 

The patch didn’t work :disappointed:

**Comment by bfontaine** [comment]
Added a new patch as suggested by upstream.

**Comment by bfontaine** [comment]
It worked! \o/

**Comment by dunn** [comment]
🎉🎊✨💥🎳

**Comment by luzpaz** [comment]
:+1:

**Comment by mistydemeo** [comment]
:shipit

**Comment by tdsmith** [comment]
Will this break things in homebrew-x11?

**Comment by DomT4** [comment]
Great to get this done :+1:. Only thing in X11 that uses `poppler` by default is `xournal`.

**Comment by bfontaine** [comment]
Yep. Stuff we need to check:

- [ ] xournal
- [ ] inkscape
- [ ] gimp

(sorry @luzpaz, we’re still not done here. But we’re progressing!)

**Comment by bfontaine** [comment]
@DomT4 Don’t we need to check stuff that don’t use poppler by default?

**Comment by DomT4** [comment]
We don't usually check non-default options. If someone reports them afterwards we look at it, but we don't usually have the resources to manually check everything non-mandatory. If you want to though, I've no objections.

File: Library/Formula/pdf2htmlex.rb [hunk 1]
```diff
@@ -20,6 +19,11 @@ class Pdf2htmlex < Formula
     url "https://github.com/coolwanglu/fontforge.git", :branch => "pdf2htmlEX"
   end
 
+  resource "AdobeXMLFormsSamples.pdf" do
+    url "http://partners.adobe.com/public/developer/en/xml/AdobeXMLFormsSamples.pdf"
+    sha256 "36547f22533af1d676e8beb4de56a206fff397481bf95fc9c95924cd8184ff8a"
+  end
```
Comment (xu-cheng) [inline Library/Formula/pdf2htmlex.rb]: Can it be replace with the pdf file stored in test features? Or it has some unique requirement.

Comment (bfontaine) [inline Library/Formula/pdf2htmlex.rb]: Yep, done.

**Comment by bfontaine** [comment]
xournal & inkscape build fine, @DomT4 is checking for gimp but poppler is an optional dependency so it’s not blocking here. If nobody objects I’ll merge this when the bot will be done.

**Pull request closed without merging**